mono/mini/exceptions-x86.c

```diff
@@ -892,7 +892,7 @@ mono_arch_unwind_frame (MonoDomain *domain, MonoJitTlsData *jit_tls,
 		else
 			/* the lmf is always stored on the stack, so the following
 			 * expression points to a stack location which can be used as ESP */
-			new_ctx->esp = (unsigned long)&((*lmf)->eip);
+			new_ctx->esp = (gsize)&((*lmf)->eip);
```

Comment:
There are other places in this file where esp is cast to different types, shouldn't they also get cleaned up?

Reply:
This is enough to get it to compile in CI.
Which was my first and perhaps last goal.

It is likely that others are masked by `#ifndef DISABLE_JIT`, for better and worse.
Worse because you get less checking in other configurations.
Better because the code might really not make sense and be dead.

I could also try to add `#ifndef DISABLE_JIT` if that is preferred.
I did some, but it isn't my preference -- avoid `#if`.

Consistency might also be good yes.
So people know what patterns to follow.
The pattern is "always cast non-pointer and pointer using same sized types (gsize/gssize/intptr_t/uintptr_t/size_t/ptrdiff_t and pointer); if you want to change size, you can do that amongst integer types).

Reply:
I added two more casts, but they are probably `#if` out.